lib/shared-method.js

```diff
@@ -137,7 +137,9 @@ function SharedMethod(fn, name, sc, options) {
     this.errors = [this.errors];
   }
 
-  this.stringName = (sc ? sc.name : '') + (isStatic ? '.' : '.prototype.') + name;
+  var addPrototypeString = !/^prototype\./.test(name);
+  this.stringName = (sc ? sc.name : '') +
+    (!isStatic && addPrototypeString ? '.prototype.' : '.') + name;
```

Comment:
ditto. Why would you like to support `new SharedMethod(fn, 'prototype.name', ...)`?